src/components/FocusBoundary.js

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+
+function isChildOf(target, child) {
+  if (!child) {
+    return false;
+  }
+
+  return target === child.parentElement
+    ? true
+    : isChildOf(target, child.parentElement);
+}
+
+export const FocusBoundary = ({ children, onChange, onEnter, onExit }) => {
+  const [ref, setRef] = React.useState();
```

Comment:
Storing this ref here as state, because we want to trigger a side-effect (adding/removing event listeners) whenever the ref changes. Storing it in state (by passing `ref={setRef}` as a prop to the child) means we can pass it to the `useEffect` dependencies array, and do stuff when it changes!